modules/common/topology/types.go

```diff
@@ -0,0 +1,33 @@
+/*
+Copyright 2025 Red Hat
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package topology
+
+// Ref - it models a Topology reference and it can be included in the
+// service operators API. It is used to retrieve the referenced Topology
+type Ref struct {
```

Comment:
@olliewalsh @stuggi how this approach is? I'm not sure we are in agreement and I'd like to get your feedback to finalize this common content.
In glance this could be referenced with:

```
// +kubebuilder:validation:Optional
// TopologyRef to apply the Policy defined by the associated CR referenced  by name
TopologyRef *topology.Ref `json:"topologyRef,omitempty"`
```

See [1] for more details.

[1] https://github.com/openstack-k8s-operators/glance-operator/pull/670/files

Reply:
lgtm

Reply:
Naming it just 'Ref' leaves me feeling a bit queasy. I concede it looks ok when prefaced with the package name (topology.Ref) but I'm curious what others thing about naming it something like 'TopoRef'. But not a blocker for me.

Reply:
@ASBishop I understand your feeling about this, and the conversation we had was based on [1]. We can decide to not follow [1], and remove `golint`, but I feel like we should find and agreement that makes us happy and at the same time does not conflict with [1].
`TopoRef`, as struct name, seems to meet these requirements because I'm ok with that and looks like golint doesn't throw an error.
If @stuggi @olliewalsh agree I would be happy to rename this struct as per @ASBishop  suggestion.

[1] https://google.github.io/styleguide/go/decisions#repetition

Reply:
works for me

Reply:
Done